Redesigning both Theme and Skin icons

now-im opened the pull request:
I have changes the themes icon in this commit. Kindly, review it and see if it is suitable for the application.

Previous:
![Screenshot_20200313_145654](https://user-images.githubusercontent.com/22009014/76651812-7839bd00-658f-11ea-877c-a5ff2e3bacc5.png)
Current:
![image](https://user-images.githubusercontent.com/22009014/76651924-b59e4a80-658f-11ea-9d1e-35cca2a0f417.png)
Note that, even after making some changes it is not showing it in preview. So, end result might be a bit different.

thalieht commented:
`edit-clear-history` and the other 4 like it, look unchanged in qBt, just like in the after image, although i can see the change on github... Also the 2 `media-...` icons on the bottom right have very little contrast, white on gray.
I hadn't realized until now that i was ignoring all these icons because the ones currently used in qBt are barely visible on a dark theme. What a treat! They also look better in light themes too (IMO).

There is one minor thing that i'm not sure it's your problem... This color is dangerously close to the one used in Breeze theme for checked checkboxes, note Automatic Torrent Management:
![Annotation 2020-03-13 220559](https://user-images.githubusercontent.com/6451685/76655826-e4f48d80-6576-11ea-8081-e6d68787f38e.png)

now-im commented:
@thalieht I have fixed those icons. I can't design the checked boxes they are done by the system theme. I have used `la-capitaine-icon-theme` for the whole project. Only cookies and bug icon has been taken from `papirus-icon-theme`. 
I have also added a guideline so that in the future it is easy to design and maintain the icons by others. I hope this gets merged.

thalieht commented:
>I can't design the checked boxes they are done by the system theme.

I know, just mentioned it in case it was annoying for someone.

thalieht commented:
The ratio icon is the old one because the new is in the other PR.

![light1](https://user-images.githubusercontent.com/6451685/76681428-48c09a00-65fb-11ea-9058-af039a1c483e.png) 
![dark1](https://user-images.githubusercontent.com/6451685/76681445-71e12a80-65fb-11ea-9630-99c3abcfc578.png)
<img src="https://user-images.githubusercontent.com/6451685/76681439-5aa23d00-65fb-11ea-9ca4-c6a11cdb1b31.png" width="80"><img src="https://user-images.githubusercontent.com/6451685/76681449-760d4800-65fb-11ea-8f0f-6c405dd3333f.png" width="80">

FranciscoPombal commented:
My thoughts:

- Shouldn't the force resume button be the same color as the normal resume button, but with 2 arrows?
- I think the force recheck should be different from the force reannounce button. The force reannounce button should be something like a "syncing" button (something like this, 3rd row, 2nd column, but more round: https://image.freepik.com/free-vector/gui-buttons-icons-set-game-interfaces-color-blue_38100-187.jpg)

now-im commented:
@FranciscoPombal I will make the necessary changes but I will not be able to do anything before 26 March. In the meantime find any other problem you can find. The icon is same because in configuration file it is same. It will take some time to find the icon name and create a new one. If you can provide the list of repeated name it will make thw job easier for me.

@thalieht I can never thank you enough for building the application again and again so that I may have a look. I had no idea it would improve the look and feel of qbittorrent this much. Thank you again.

now-im commented:
@FranciscoPombal There may not be a separate icon for the force renounce. That is why it is adopting another icon. Give me a name which I can make an SVG of I will do it. I also want to make Tag icon. It would be nice.

I have made all folder icon to make them homogenous.

now-im commented:
@thalieht Anymore improvement suggestions? 

@Chocobo1 @sledgehammer999 @glassez 
I have worked hard on this project despite having a full-time job. It is really disappointing that I don't get any feedback from maintainer and developers. I still don't know if the changes are going to be merged or not. Kindly give me some form of an update.

thalieht commented:
Resume icon seems to be transparent unlike Pause and Force Resume. Also maybe it's just me but it's hard for me to distinguish what's inside the Force Resume icon.
The following is my opinion and i don't have any suggestions so feel free to ignore it: I don't like how these 3 icons are in a circle because it makes them even smaller.

now-im commented:
@thalieht You were right from design perspective and usability those did not make any sense. I removed the circles and changed colors. I have made some other changes and all my commits are here now. Can you kindly compile one last time and give me a final look?
I am closing two other pull requests (#8406 and #12276) as I have added all my changes here. I don't think I need to make any more improvements.

thalieht commented:
![light](https://user-images.githubusercontent.com/6451685/77705318-023e4880-6fc8-11ea-9a48-eb8dc50905bd.png)
![dark](https://user-images.githubusercontent.com/6451685/77705314-ff435800-6fc7-11ea-9fc9-416b6b2a993a.png)

👍

FranciscoPombal commented:
@now-im This looks really good right now. I have no complaints. @thalieht can you please post more screenshots, from settings, etc?

EDIT: except 2 minor details:
- Why do "categories" and "tags" icons in the right click menu turn black in the dark theme, but "rename" and "edit trackers" don't
- the firewalled icon is still a shield, which still conveys the meaning of protection. How about a flame with an exclamation mark instead?

thalieht commented:
>@thalieht can you please post more screenshots, from settings, etc?

They're in https://github.com/qbittorrent/qBittorrent/pull/12169#issuecomment-599047882  except the BitTorrent gear is orange now.

Also
<img src="https://user-images.githubusercontent.com/6451685/77710013-8ea33800-6fd5-11ea-8330-29b2aa92876b.gif" width="300">

FranciscoPombal commented:
@now-im 

IMO I'd either leave the connections icon in the settings as the ethernet port or I'd change to a wifi symbol as well.

Can't say I am a fan of the Bittorrent icon in the settings, either. I am thinking something like drawing 2 or 3 from here (https://www.rs-online.com/designspark/rel-assets/dsauto/temp/uploaded/network_types.png?w=815), but less symmetric and with less nodes and less lines also in the case of 3 so that the point comes across without looking too small/cluttered.

But it's up to you, since you've already done so much.

now-im commented:
@FranciscoPombal I got two icons. See which feels right.
![image](https://user-images.githubusercontent.com/22009014/77726489-3181a380-7022-11ea-9f85-6141771b5f2c.png)
![image](https://user-images.githubusercontent.com/22009014/77726557-65f55f80-7022-11ea-89cd-7aee507d10aa.png)

FranciscoPombal commented:
@now-im 

> @FranciscoPombal I got two icons. See which feels right.

I vote for the first one, but I'd wait for more opinions @thalieht.

> On the other note, I think changing the donate Icon from wallet to love will be a good idea.

Yes, good idea.

> I have also changed the magnet icon from orange to green. Will this work?

Now that you mention it, I don't think any icons other than those associated with "activity"/"downloading"/"resuming" should be green. The only icons that need changing are the "documentation" and "about", from green to blue. also shouldn't the "about" icon be different? something like this: http://icon-library.com/images/about-icon/about-icon-25.jpg

About the magnet icon specifically, I think it works as it was (orange), but blue like the other ones also would be good. I'll leave the final choice of the magnet icon color up to you.

now-im commented:
> The only icons that need changing are the "documentation" and "about", from green to blue. also shouldn't the "about" icon be different?

Here is the problem I am facing. Right now there are four icons that have a question mark in them. `dialog-information.svg, help-about.svg and help-contents.svg`. I got no idea which is the one represents about. I have no problem changing it to an "i". 



> About the magnet icon specifically, I think it works as it was (orange), but blue like the other ones also would be good. I'll leave the final choice of the magnet icon color up to you.

Then I am keeping it as orange. It just pops out among other icons.

FranciscoPombal commented:
> Here is the problem I am facing. Right now there are four icons that have a question mark in them. `dialog-information.svg, help-about.svg and help-contents.svg`. I got no idea which is the one represents about. I have no problem changing it to an "i".

it's `help-about.svg`. `dialog-information.svg` should also be an `i` (can be a different one, up to you) if it isn't already.

> Then I am keeping it as orange. It just pops out among other icons.

:+1: 
 
> P.S. Only 3 icon was using yellow color. I have replaced them with orange. Less color clutter this way.

:+1:

now-im commented:
@FranciscoPombal Then I think my work here is done. I have pushed all changes. If qbittorent project ever needs a new app icon, knock me. I will do whatever I can.

FranciscoPombal commented:
@now-im 
> @FranciscoPombal Then I think my work here is done. I have pushed all changes. If qbittorent project ever needs a new app icon, knock me. I will do whatever I can.

Don't forget to change the "about"/"information" icons to `i`'s

PS - your comment shows up as edited because I accidentally clicked that instead of "quote reply", but I reverted the changes I made

now-im commented:
> @now-im
> 
> > @FranciscoPombal Then I think my work here is done. I have pushed all changes. If qbittorent project ever needs a new app icon, knock me. I will do whatever I can.
> 
> Don't forget to change the "about"/"information" icons to `i`'s
> 
I have made the changes in `Improving Donation, Magnet, Help icons` commit. I had a hunch. You confirmed it in a later comment. So, it is already done.
![image](https://user-images.githubusercontent.com/22009014/77769050-88609a80-706d-11ea-876b-f372529cdf5a.png)
This is `dialog-information.svg` now.

FranciscoPombal commented:
My bad, didn't notice.

Outstanding job. I have no further comments.

@glassez @Chocobo1 @sledgehammer999 review for merging please.

glassez commented:
Personally, I don't mind updating the app's appearance from time to time.
As for the icons themselves, this is subjective... I prefer not to engage in arguments about subjective topics. So if you approve some icon set by a majority vote and, most importantly, if the maintainer also approves it, I don't mind.
The only objection is that we should not apply such changes between minor releases.

glassez commented:
Also

Chocobo1 commented:
A few thoughts:
* @now-im:
  * I noticed there are some files such as src/icons/skin/downloading.svg.orig, src/icons/skin/downloading_BACKUP_7148.svg, are those really in use? I suppose you commit them accidentally?
  * There is a .svgo.yml in the icons folder, you should use it to slim down the svg files when everything is settled.
  ps. don't use other tools, IIRC it will destroy the viewbox or some other important property in the svg file.
  * There are a few newly added icons, which I believe are not currently in use... 

* Anyone interested in reviewing src/icons/skin/Design Guidelines.md?

* @thalieht What about WebUI? Does it look nice too?

now-im commented:
> Also you need to squash all your commits into one and fix commit message (and PR title) before we can merge it.

Done.

>     * I noticed there are some files such as src/icons/skin/downloading.svg.orig, src/icons/skin/downloading_BACKUP_7148.svg, are those really in use? I suppose you commit them accidentally?

They were in the main repository so I changed them too. I have no idea what .orig extension does.



>     * There is a .svgo.yml in the icons folder, you should use it to slim down the svg files when everything is settled.
>       ps. don't use other tools, IIRC it will destroy the viewbox or some other important property in the svg file.

Icons from both theme and skin folders have an accumulated size under 600kb. I think leaving the icon as is will help people who want to modify them in the future. However, I will do it if you insist.



>     * There are a few newly added icons, which I believe are not currently in use...

I just modified available icons. Added nothing, deleted nothing. I would like to add "Tags" icon if anyone can provide me the SVG file name.

FranciscoPombal commented:
@now-im 
> They were in the main repository so I changed them too. I have no idea what .orig extension does.

Something in your setup must have created them accidentally, they are not present in the current repo. The `.orig` extension typically means a "backup".

thalieht commented:
>@thalieht What about WebUI? Does it look nice too?

The same

![webui](https://user-images.githubusercontent.com/6451685/77838170-fdb99180-7179-11ea-9608-c48e6c194e3d.png)

@now-im are these the correct icons for `Documentation` and `About`?

![em](https://user-images.githubusercontent.com/6451685/77838189-3bb6b580-717a-11ea-8159-bfa9cb7e30fa.png)

>I would like to add "Tags" icon if anyone can provide me the SVG file name.

Whatever you see fit? or i didn't understand the question.

now-im commented:
> @now-im are these the correct icons for `Documentation` and `About`?

I have changed them into this now. 
![image](https://user-images.githubusercontent.com/22009014/77841003-18f2c400-71af-11ea-873e-636d105a57d2.png)

> Whatever you see fit? or i didn't understand the question.
I want to change these two pairs of same icons into a different one. Somehow they are linked as same. Can you give me some directions?

![77838170-fdb99180-7179-11ea-9608-c48e6c194e3d](https://user-images.githubusercontent.com/22009014/77841011-2c059400-71af-11ea-8748-714651a3cbb5.png)



> Something in your setup must have created them accidentally, they are not present in the current repo. The `.orig` extension typically means a "backup".

Okay, I have removed them. I had no idea how I created them,

FranciscoPombal commented:
@now-im 
Ok, I think the tag and category icons can be the same, but now I see a few issues with the WebUI screenshots I hadn't noticed before:
- The "force reannounce" icon should be different than the force recheck icon; it should be something like a "sync" icon (example: https://cdn3.iconfinder.com/data/icons/faticons/32/sync-01-512.png)
- There are no icons for "queue", "download first and last pieces first", and "download in sequential order" in that context menu. Is this intentional?

Sorry for insisting for more changes when I had already indicated that I had nothing more to say.

ArcticGems commented:
>     * The "force reannounce" icon should be different than the force recheck icon; it should be something like a "sync" icon (example: https://cdn3.iconfinder.com/data/icons/faticons/32/sync-01-512.png)
> 

Or something like this? =
https://www.iconfinder.com/icons/115884/delay_reload_time_icon

FranciscoPombal commented:
> Or something like this? =
> https://www.iconfinder.com/icons/115884/delay_reload_time_icon

Disagree, it has nothing to do with time or time intervals.

ArcticGems commented:
What about a simple bell with refresh arrows in/on it (icon below doesn't have refresh arrows)? =
https://www.iconfinder.com/icons/452625/announce_bell_ring_sound_icon

xavier2k6 commented:
The "delete icon" looks like a glass with two ice cubes in it to me anyway.

thalieht commented:
>I want to change these two pairs of same icons into a different one. Somehow they are linked as same. Can you give me some directions?

It's easy you just need to change "view-categories" in 1 line in 2 files to the new file name, also to add an entry with the new file in `src/icons/icons.qrc`, note the alphabetical order.
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistwidget.cpp#L1223
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/webui/www/private/index.html#L140

On another note i saw that "edit-delete" icon is the recycle one but in qBt all deletes are final.

now-im commented:
> > Or something like this? =
> > https://www.iconfinder.com/icons/115884/delay_reload_time_icon
> 
> Disagree, it has nothing to do with time or time intervals.

It looks like this now. Previous icon was complicated for a small icon.
![image](https://user-images.githubusercontent.com/22009014/77849725-94746580-71ef-11ea-982b-24e63a29be41.png)

xavier2k6 commented:
>On another note i saw that "edit-delete" icon is the recycle one but in qBt all deletes are final.

Maybe it needs to be changed to just a red "X"

now-im commented:
> It's easy you just need to change "view-categories" in 1 line in 2 files to the new file name, also to add an entry with the new file in `src/icons/icons.qrc`, note the alphabetical order.
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistwidget.cpp#L1223





> > I want to change these two pairs of same icons into a different one. Somehow they are linked as same. Can you give me some directions?
> 
> It's easy you just need to change "view-categories" in 1 line in 2 files to the new file name, also to add an entry with the new file in `src/icons/icons.qrc`, note the alphabetical order.
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistwidget.cpp#L1223
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/webui/www/private/index.html#L140
> 
> On another note i saw that "edit-delete" icon is the recycle one but in qBt all deletes are final.

I have made a commit dc9386c for the Tags icon. Kindly review it. I am working on the delete icon.

Chocobo1 commented:
> > Something in your setup must have created them accidentally, they are not present in the current repo. The `.orig` extension typically means a "backup".
> 
> Okay, I have removed them. I had no idea how I created them,

But there are more of them.
You can find them under this link: https://github.com/qbittorrent/qBittorrent/pull/12169/files and click on the "Jump to ..." label. It will display a list of files, notice the first icon will be green if that file is newly added. Please make sure there are no newly added files.

thalieht commented:
>I have made a commit dc9386c for the Tags icon. Kindly review it.

Looks good 👍 

![tags](https://user-images.githubusercontent.com/6451685/77850325-5fa9e380-71da-11ea-8d8c-26ab965bdcef.png)

now-im commented:
> > > Something in your setup must have created them accidentally, they are not present in the current repo. The `.orig` extension typically means a "backup".
> > 
> > 
> > Okay, I have removed them. I had no idea how I created them,
> 
> But there are more of them.
> You can find them under this link: https://github.com/qbittorrent/qBittorrent/pull/12169/files and click on the "Jump to ..." label. It will display a list of files, notice the first icon will be green if that file is newly added. Please make sure there are no newly added files.

I removed only the .orig icons previously. Thanks for the direction. Made necessary changes in 35af412. `tags.svg` is a new icon. I will add another for force recheck.

now-im commented:
> > I have made a commit [dc9386c](https://github.com/qbittorrent/qBittorrent/commit/dc9386c0aee9c5f548bb4248c38e4a6a4b5db8e0) for the Tags icon. Kindly review it.
> 
> Looks good +1
> 
> ![tags](https://user-images.githubusercontent.com/6451685/77850325-5fa9e380-71da-11ea-8d8c-26ab965bdcef.png)

I have split the reannounce and recheck icon. Strangely, they used an icon named "document-edit-verify" which made no sense. I have changed them to recheck.svg and reannounce.svg.

now-im commented:
I thank everyone who has provided their take on this work. I had no idea it would improve the initial design this much. 
I wholeheartedly thank @keeferrourke, as without his [icon set](https://github.com/keeferrourke/la-capitaine-icon-theme) I could never do this, as I am not a professional graphics designer. 

You guys are awesome!

thalieht commented:
![menu](https://user-images.githubusercontent.com/6451685/77853031-7c4e1780-71ea-11ea-9e9b-02365fa82607.png)

👍 but reannounce is inconsistent with the actions in the tracker tab

![trac](https://user-images.githubusercontent.com/6451685/77853070-be775900-71ea-11ea-8cdf-1a5c26ce4d9f.png)
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/properties/trackerlistwidget.cpp#L571
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/properties/trackerlistwidget.cpp#L576

now-im commented:
> ![menu](https://user-images.githubusercontent.com/6451685/77853031-7c4e1780-71ea-11ea-9e9b-02365fa82607.png)
> 
> +1 but reannounce is inconsistent with the actions in the tracker tab
> 
> ![trac](https://user-images.githubusercontent.com/6451685/77853070-be775900-71ea-11ea-8cdf-1a5c26ce4d9f.png)
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/properties/trackerlistwidget.cpp#L571
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/properties/trackerlistwidget.cpp#L576

Thanks a lot. Pushed the changes.

glassez commented:
>>Also you need to squash all your commits into one and fix commit message (and PR title) before we can merge it.
>
>Done.

Sorry. I mean you need to fix commit message according to project Coding Guidelines.

Chocobo1 commented on src/gui/properties/trackerlistwidget.cpp:
```diff
@@ -568,12 +568,12 @@ void TrackerListWidget::showTrackerListMenu(const QPoint &)
     }
 
     if (!torrent->isPaused()) {
-        const QAction *reannounceSelAct = menu->addAction(UIThemeManager::instance()->getIcon("view-refresh"), tr("Force reannounce to selected trackers"));
```
Before you guys go all out, just keep in mind the original icon name has an advantage which is using the icon from desktop environment (if they provided one), this makes the application having more native feel. https://developer.gnome.org/icon-naming-spec/
IMO we should use those standard icon names as much as possible and only use custom ones when standard ones are not suitable.

now-im replied:
I didn't know that. Let me adjust using the icon specific name if possible.

now-im replied:
I have checked the gnome-icon-theme. There is nothing that can replace `reannounce` and `tag` icon. 
I am going to change the `recheck` icon to `view-refresh` as they are very similar. Great tip!

thalieht commented:
Anyone knows why we use style standard icons for Warning and Error in trackers filter list? I hadn't noticed it until now because it doesn't look too bad in Windows (IMO) but in Linux plasma it's completely out of place:

![trackers](https://user-images.githubusercontent.com/6451685/77853880-4bbcac80-71ef-11ea-8105-fd60f5b8ac48.png)

In case you want to do something about it:
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistfilterswidget.cpp#L290
https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistfilterswidget.cpp#L293

FranciscoPombal commented:
@now-im I believe there are still the following missing icons in the right-click context menu:
- Queue
- Automatic Torrent Management
- Super seeding mode
- Download in sequential order
- Download first and last pieces first

Plus on XFCE (Xubuntu 18.04) I still see a lot of the old icons. I would appreciate if someone could reproduce this.

thalieht commented:
>Automatic Torrent Management
>Super seeding mode
>Download in sequential order
>Download first and last pieces first

But these are checkboxes.

FranciscoPombal commented:
>>There is a .svgo.yml in the icons folder, you should use it to slim down the svg files when everything is settled.
>>
>> ps. don't use other tools, IIRC it will destroy the viewbox or some other important property in the svg file.

> Icons from both theme and skin folders have an accumulated size under 600kb. I think leaving the icon as is will help people who want to modify them in the future. However, I will do it if you insist.

Are designers actually expected to edit stuff in the svg files by hand (I really don't know)? If not, I think you should minify them like @Chocobo1 says. It's not about disk space used IMO, but the extra size adds up and is undesirable in the context of the size of requests to the WebUI, especially the initial one. This won't be visible in fast and reliable connections, but I think the requests to the WebUI should be as lean as possible.

FranciscoPombal commented:
@thalieht @now-im 
> But these are checkboxes.

Oh yeah. Turns out on my system, unchecked checkboxes are just blank space, which makes it seem like there is something missing. Disregard my previous remark about this then. Still, is this expected behavior (that on some systems unchecked checkboxes are completely blank)?

FranciscoPombal commented:
Minor nitpick: I think the flame icon should be simpler, more like the shape of this one (but still with the exclamation mark of course): https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/flame.png

now-im commented:
> In case you want to do something about it:
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistfilterswidget.cpp#L290
> 
> https://github.com/qbittorrent/qBittorrent/blob/8a8607bf93bb0bdc0ffa533847a96c762b0ed9f9/src/gui/transferlistfilterswidget.cpp#L293
I have pushed some changes, see if that works.

> Are designers actually expected to edit stuff in the svg files by hand (I really don't know)? If not, I think you should minify them like @Chocobo1 says. It's not about disk space used IMO, but the extra size adds up and is undesirable in the context of the size of requests to the WebUI, especially the initial one. This won't be visible in fast and reliable connections, but I think the requests to the WebUI should be as lean as possible.

Trust me on this one. With today's internet speed it won't be a problem. By saving let's say 400kb we are forgoing metadata that can't be recovered. Shapes inside the icon become fixed and can't be improved upon.




> Oh yeah. Turns out on my system, unchecked checkboxes are just blank space, which makes it seem like there is something missing. Disregard my previous remark about this then. Still, is this expected behavior (that on some systems unchecked checkboxes are completely blank)?

That is because of the theme you are using. Changing it may help.



> Minor nitpick: I think the flame icon should be simpler, more like the shape of this one (but still with the exclamation mark of course): https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/flame.png

Let's see what I can do.

now-im commented:
> Minor nitpick: I think the flame icon should be simpler, more like the shape of this one (but still with the exclamation mark of course): https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/flame.png

Will this work?

![image](https://user-images.githubusercontent.com/22009014/77855210-cc3fd500-7210-11ea-878f-23e46520efbc.png)
or,
![image](https://user-images.githubusercontent.com/22009014/77855254-07da9f00-7211-11ea-8466-6733ac275c1c.png)

thalieht commented on src/gui/transferlistfilterswidget.cpp:
```diff
@@ -287,10 +287,10 @@ TrackerFiltersList::TrackerFiltersList(QWidget *parent, TransferListWidget *tran
     auto *errorTracker = new QListWidgetItem(this);
     errorTracker->setData(Qt::DisplayRole, tr("Error (0)"));
-    errorTracker->setData(Qt::DecorationRole, style()->standardIcon(QStyle::SP_MessageBoxCritical));
+    errorTracker->setData(Qt::DecorationRole, style()->getIcon("error"));
```
Ah sorry these are not the same like the previous ones:
```suggestion
    errorTracker->setData(Qt::DecorationRole, UIThemeManager::instance()->getIcon("error"));
```

now-im commented:
@glassez Will d508327 work?

FranciscoPombal commented:
> Trust me on this one. With today's internet speed it won't be a problem. By saving let's say 400kb we are forgoing metadata that can't be recovered. Shapes inside the icon become fixed and can't be improved upon.

Maybe icons with metadata can be kept in a separate folder that's not used for building, and minified icons can be used like before. But don't make this change now, wait for the opinions of @sledgehammer999 @Chocobo1 @glassez on this one.

> > Oh yeah. Turns out on my system, unchecked checkboxes are just blank space, which makes it seem like there is something missing. Disregard my previous remark about this then. Still, is this expected behavior (that on some systems unchecked checkboxes are completely blank)?
> 
> That is because of the theme you are using. Changing it may help.

I see. Are you sure it does not have to do with non-standard icon names or something? The theme I'm using is Greybird, the default for Xubuntu.

> > Minor nitpick: I think the flame icon should be simpler, more like the shape of this one (but still with the exclamation mark of course): https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/flame.png
> 
> Let's see what I can do.

Don't take this personally, but it looks a bit like a poop. Maybe it needs to be a little less "fat around the base"? If that makes sense

now-im commented:
> I see. Are you sure it does not have to do with non-standard icon names or something? The theme I'm using is Greybird, the default for Xubuntu.
I have only removed one icon. The others are left as is. I think the problem is either with the theme or the icon pack you are using. La-Capitane-icon-theme or Papirus-icon-theme is what I use.

> Don't take this personally, but it looks a bit like a poop. Maybe it needs to be a little less "fat around the base"? If that makes sense

I am not good at making complicated icons. I just can modify them. I have taken one icon from [Simple Icon](https://simpleicons.org/) to make this one.

![image](https://user-images.githubusercontent.com/22009014/77857093-5ee67100-721d-11ea-8df4-3cce49d21eb2.png)

ArcticGems commented:
> 
> I am not good at making complicated icons. I just can modify them. I have taken one icon from [Simple Icon](https://simpleicons.org/) to make this one.
> 
> ![image](https://user-images.githubusercontent.com/22009014/77857093-5ee67100-721d-11ea-8df4-3cce49d21eb2.png)

Much better! :smiley:

FranciscoPombal commented:
> I am not good at making complicated icons. I just can modify them. I have taken one icon from [Simple Icon](https://simpleicons.org/) to make this one.
> 
> ![image](https://user-images.githubusercontent.com/22009014/77857093-5ee67100-721d-11ea-8df4-3cce49d21eb2.png)

This one is great!

NotTsunami commented:
I recommend running the images through a minimizer as well, such as [this](https://vecta.io/nano) one.

now-im commented:
Ok, Based on the popular opinion I have created a backup and made optimization based on the .svgo.yml config file available. The file size became 230kb from 600kb. This is the end result:
![image](https://user-images.githubusercontent.com/22009014/77886827-5d539200-728b-11ea-992d-450d640cb3cf.png)
The interesting thing is when I open the icon it is ok. But the preview is totally busted. Like this:
Preview:
![image](https://user-images.githubusercontent.com/22009014/77887016-aa376880-728b-11ea-9052-c1b49d5aaac3.png)
In Inkscape:
![image](https://user-images.githubusercontent.com/22009014/77887050-b7ecee00-728b-11ea-90eb-04683bbde395.png)
Now, the decision is up to you. I have made my opinion clear on previous comment.

NotTsunami commented:
Does the same happen if you run them through [this](https://vecta.io/nano) site?

NotTsunami commented:
I'm gonna play around with the included svgo config and see if it can be improved or that thumbnail issue can be fixed.

now-im commented:
> Does the same happen if you run them through [this](https://vecta.io/nano) site?

I have had previous experience with [svgcleaner](https://github.com/RazrFalcon/SVGCleaner). Using it yielded a better result. 180kb without icon breakage. 
![image](https://user-images.githubusercontent.com/22009014/77889821-4bc0b900-7290-11ea-9f98-64a692114f60.png)

@Chocobo1 @FranciscoPombal What do you think? I am still against it but at least this program doesn't cause icon breakage.

FranciscoPombal commented:
@now-im Like I said, if it were up to me we could do one of the following:

1. keep both sets (optimized and non-optimized) in the repository.
    - The increase in size in the repository is irrevelevant
    - Users benefits still benefit from using the optimized icons during program execution (release builds would use the optimized versions)
    - Developers/designers benefit from the existence of the non-optimized versions for future improvements, like you said

2. keep only the non-optimized version and make the svg optimizer pass part of the build process, but this would mean:
    - another build step, possibly optional (e.g. controlled by a flag such as `--enable-svg-optmization`/`-DSVG-OPTIMIZATION`) and for this build step either:
        - An extra build dependency
        - Requirement to be online and use an online service (I would really dislike this, even considering that for development purposes it would be ok to just use the non-optimized versions)

If option 2 is chosen, this PR would be merged with only the non-optimized icons and the build system changes stuff handled in another one. It is overall the better option IMO, since it is natural to think about non-optimized svg files as "sources" and optimized svg files as "build products". However, it is a little more complex, as it involves changes to the build system as a follow-up to this PR.

Either way this kind of decision is up to @Chocobo1 @glassez @sledgehammer999

Chocobo1 commented:
>The interesting thing is when I open the icon it is ok. But the preview is totally busted. Like this:

Now I recall such moments...

> I have had previous experience with svgcleaner. Using it yielded a better result. 180kb without icon breakage.

👍  According to their website they should have better correctness (and written in Rust should be a plus too IMO). We probably should ditch svgo.

@now-im 
I won't insist you do the svg optimization in this PR. If you don't, I would still optimize it later in another PR. And if you want to do it, then please make it another commit. At least that way we still have the original files in history.
And how does using svgcleaner gets in the way of editing? (not a designer here) Are you using layers or something else and it will be modified?

now-im commented:
> I won't insist you do the svg optimization in this PR. If you don't, I would still optimize it later in another PR. And if you want to do it, then please make it another commit. At least that way we still have the original files in history.
> And how does using svgcleaner gets in the way of editing? (not a designer here) Are you using layers or something else and it will be modified?

Previously, they would combine different shapes into one and thus making the icon hard for modification. Now, after tinkering with the icon I can see that they have sloved the problem. Let me open a new PR which includes the icon before the optimization.

jagannatharjun commented:
just sharing my opinion .....
imho the icons are too bright, can't we use the same shade of blue that is used in qbittorrent's icon?

jagannatharjun commented:
>  I have kept the original commit untouched. I think this is ready for the merge.


I think maintainers would prefer to keep the history in the official repository, since chocobo1 agrees you don't need to optimize anyway and 300 kb is nothing if you see the target audience of qbittorrent.

jagannatharjun commented:
how about making them the same size as original ones, also see the rss icon in tab
![image](https://user-images.githubusercontent.com/34717789/77904521-4bc9b480-72a2-11ea-9028-f31c71860af0.png)

I think this status bar icon needs some work
![image](https://user-images.githubusercontent.com/34717789/77904744-ae22b500-72a2-11ea-924c-afc7e00473aa.png)


also don't you guys think the GUI looks too much colorful now (at least on windows), I personally don't like it, anyone window here? please share your opinions.
![image](https://user-images.githubusercontent.com/34717789/77904605-73b91800-72a2-11ea-942a-62c6c584ca38.png)

jagannatharjun commented:
It does look better on the dark theme though, I think you need to find a balance
![image](https://user-images.githubusercontent.com/34717789/77906231-3a35dc00-72a5-11ea-8547-b114a8502795.png)

now-im commented:
That was the main reason for the redesign. Most of this icon is barely visible in the dark themes. Besides, the most modern application is embracing the dark themes and bright colors. I will leave it up to the majority. This is my take.

jagannatharjun commented:
everything else is good and acceptable but just that the top bar items really look out of order if you can decrease their heights, maybe match with the original icons that it will be really good.

now-im commented:
Here is my proposal, I can use the color used by google. It should work in both light and dark situations. 
![image](https://user-images.githubusercontent.com/22009014/77920712-a245eb80-72c0-11ea-9fca-374d7c3a750a.png)
Or the one used by Microsoft,
![image](https://user-images.githubusercontent.com/22009014/77920804-c1447d80-72c0-11ea-9dd6-68d7fef20651.png)



> everything else is good and acceptable but just that the top bar items really look out of order if you can decrease their heights, maybe match with the original icons that it will be really good.

let's see what I can do.

now-im commented:
let's take some opinion before that. I have had to redesign this too many times. 
@thalieht @FranciscoPombal @Chocobo1 So you agree with @jagannatharjun?

FranciscoPombal commented:
@jagannatharjun 
> how about making them the same size as original ones, also see the tab icons needs to be bit small

> I think this status bar icon needs some work, maybe decrease their heights a bit, also see the alternate icon button

Hmm, agreed.

> also don't you guys think the GUI looks too much colorful now (at least on windows), I personally don't like it that much, anyone window user here? please share your opinions.

I don't think the number of different colors used needs to be decreased, but maybe the colors do need to be a less bright and/or saturated.

jagannatharjun commented:
> I don't think the number of different colors used needs to be decreased, but maybe the colors do need to be a less bright and/or saturated.

windows Native widgets are somewhat old, and too many colors on it look distracting, I personally like the original icons and icon choices and the only problem seems with them is too dim coloring. 
The icons of this PR are also very good and definitely an improvement but I don't like some choices especially sizes of icons, I would really like it if the sizes of original icons can be preserved. but this is only my personal opinion

jagannatharjun commented:
> Here is my proposal, I can use the color used by google. It should work in both light and dark situations.

I also think you should go with google, google design guidelines seem better, Microsoft is always confused.

Chocobo1 commented:
>@thalieht @FranciscoPombal @Chocobo1 So you agree with @jagannatharjun?

I prefer not state my personal opinion. However, good looking or not is purely subjective to each one, and refacing an app always attracts positive and negative comments, for example I can already see dissatisfied users want the original icons back or having an option to change as they wish (there is a related PR in the work IIRC). Just keep this in mind when facing users filing this kind of issues (hopefully there won't be many).

now-im commented:
@Chocobo1 @FranciscoPombal @jagannatharjun 
I am changing the color based on google guideline. Google doesn't have orange in so I can either use the yellow they use or change it to the orange used by Microsoft. This is consuming a lot of my time. Let's do this right once and for all.

> > how about making them the same size as original ones, also see the tab icons needs to be bit small
> 
> > I think this status bar icon needs some work, maybe decrease their heights a bit, also see the alternate icon button
> 
> Hmm, agreed.
Previous icons didn't have a fixed canvas or icon size. That is why my icons are very close to the size of others. I am resizing the icon but I will still try to keep them in same size. This is good practice.

@Chocobo1 Should I optimize or not? The existing npm based `svgo ` is not a good piece of software. It breaks icons. I have used `svgcleaner` instead and it is reliable. It also preserves some metadata.

Before optimization: 486 kb
After optimization: 180kb

FranciscoPombal commented:
@now-im I agree with @jagannatharjun that the icons should retain their sizes, but other than that, I can leave the discussion about colors to others, as it doesn't bother me either way.

FranciscoPombal commented:
@now-im keep it simple, just leave this PR without optimization, then submit another with optimization. Once both versions are in our git history, in the future a 3rd PR can be made to add the SVG optimization as a (optional) build step.

now-im commented:
@FranciscoPombal I have resized most of them. There are two settings icon. One is `gear.svg` another is `gear32.svg`. Can you tell me which one is being used here?
https://user-images.githubusercontent.com/34717789/77904521-4bc9b480-72a2-11ea-9028-f31c71860af0.png

FranciscoPombal commented:
> @FranciscoPombal I have resized most of them. There are two settings icon. One is `gear.svg` another is `gear32.svg`. Can you tell me which one is being used here?
> https://user-images.githubusercontent.com/34717789/77904521-4bc9b480-72a2-11ea-9028-f31c71860af0.png

No idea right now. If no one knows, just replace each one at a time with a placeholder, such as a black circle/star/whatever and see where it ends up in each case.

now-im commented:
@FranciscoPombal I have changed icon size and color. This PR is unoptimized. 
I have created a new one where it is optimized(#12337).

thalieht commented:
>There are two settings icon. One is gear.svg another is gear32.svg. Can you tell me which one is being used here?

Neither, `configure.svg` is used. gear and gear32 don't seem to be used anywhere.

jagannatharjun commented:
much better may bit smaller  pixel square or two

![image](https://user-images.githubusercontent.com/34717789/77944354-746ea000-72dc-11ea-8ca2-466248b35ff4.png)

now-im commented:
> Neither, `configure.svg` is used. gear and gear32 don't seem to be used anywhere.

Exactly, I have searched every .cpp file in the gui folder. There is not a single mention of these two icons. Why are they in the folder then?

jagannatharjun commented:
also if you can remove dots from search icons, they're not clearly visible due to small size thus don't look good.

now-im commented:
> also if you can remove dots from search icons, they're not clearly visible due to small size thus don't look good.

Done.

Chocobo1 commented:
>Neither, configure.svg is used. gear and gear32 don't seem to be used anywhere.

Then they should be removed to slim down the qbt binary.

now-im commented:
> > Neither, configure.svg is used. gear and gear32 don't seem to be used anywhere.
> 
> Then they should be removed to slim down the qbt binary.

Done.

NotTsunami commented on src/icons/icons.qrc:
```diff
@@ -308,13 +307,15 @@
         <file>qbt-theme/preferences-other.svg</file>
         <file>qbt-theme/preferences-system-network.svg</file>
         <file>qbt-theme/preferences-web-browser-cookies.svg</file>
+        <file>qbt-theme/reannounce.svg</file>
```
Any files deleted (gear and gear32) must also be removed from this file, otherwise the compiler still looks for them.

now-im replied:
Thanks. I forgot about that file. Made necessary changes.

jagannatharjun replied:
are you sure they are not used in WebUI?

thalieht commented on src/gui/transferlistfilterswidget.cpp:
```diff
@@ -287,10 +287,10 @@ TrackerFiltersList::TrackerFiltersList(QWidget *parent, TransferListWidget *tran
     auto *errorTracker = new QListWidgetItem(this);
     errorTracker->setData(Qt::DisplayRole, tr("Error (0)"));
-    errorTracker->setData(Qt::DecorationRole, style()->standardIcon(QStyle::SP_MessageBoxCritical));
+    errorTracker->setData(Qt::DecorationRole, UIThemeManager::instance()->getIcon("error"));
```
error.svg should be in `qBt-theme` folder, not `skin` otherwise this won't work.

now-im replied:
Won't this change cause problem for other icons?

now-im replied:
Done.

thalieht replied:
You are right... you could undo what i suggested and change this line to
```suggestion
    errorTracker->setData(Qt::DecorationRole, QIcon(QLatin1String(":/icons/skin/ratio.svg")));
```
But i don't know if this will be acceptable. Better leave this comment unresolved until someone else confirms if this is the proper way.

Personally i don't understand why there are 2 folders. qBt-theme and skin. The icons contained in each seem randomly divided to me.

now-im replied:
Exactly my thought. Unifying them will actually be a really good solution. As I am not a programmer, I thought It may serve some special purposes.
@FranciscoPombal @Chocobo1 What do you guys think?

FranciscoPombal replied:
Maybe some historical reason? No idea if it is important or not. Differing to @Chocobo1

jagannatharjun replied:
I think the qbt-theme icons can be replaced with system icons on unix systems that's why there are rwo folders

now-im replied:
@jagannatharjun I don't think so. It is the icon name that matter not the folder they are in.

jagannatharjun replied:
https://github.com/qbittorrent/qBittorrent/blob/4b409700b0809f82b8aa853b5b936623100bc22a/src/base/iconprovider.cpp#L61

haven't you noticed almost all system  (or qbt-theme folders') icons are wrapped in `getIcon` call

now-im replied:
@jagannatharjun You misunderstood me. I said in order to use icon-theme used by the Linux Operating System, I don't think the folder name doesn't matter. It is the real icon name like `error.svg` plays the most part.

now-im replied:
@thalieht I will make the changes as soon as I get an instruction from @Chocobo1. Until then I think changing the folder is a better solution than changing the code.

Chocobo1 replied:
>Maybe some historical reason? 

Definitely some history in there, but I don't recall the details.

I'm not really sure what you guys are asking... about the suggested changes [here](https://github.com/qbittorrent/qBittorrent/pull/12169#discussion_r402402428), I don't think it is future-proof. There is some theming work going on and that change won't fit well, it would be better to load the icons via `UIThemeManager`.

I would also suggest less changes to the .cpp code in this PR as it gets complicated and it is easy to overlook something. If changes to .cpp are required it would be best to do it in a follow-up PR.

now-im replied:
@Chocobo1 Good point. I recommend unifying these icon in a single folder in future.
@thalieht I have found where `error.svg` is being used, by running following commands. How can we solve this now, following Mike's suggestion? 

```
grep -rnw -e 'error.svg'
transferlistfilterswidget.cpp:200:    errored->setData(Qt::DecorationRole, QIcon(":/icons/skin/error.svg"));
transferlistmodel.cpp:754:    static QIcon cached = QIcon(":/icons/skin/error.svg");

```

```
grep -rnw -e 'getIcon("error")'
transferlistfilterswidget.cpp:290:    errorTracker->setData(Qt::DecorationRole, UIThemeManager::instance()->getIcon("error"));
```

FranciscoPombal replied:
@now-im

I would now recommend just focusing on the icons themselves for this PR. That's done and can be delivered.

If some icons keep strange names like "document-edit-verify" or if they are split across folders seemingly arbitrarily, that's something for another issue report/PR in the future, separate from this one.

now-im replied:
@FranciscoPombal I just want to fix just `error.svg` icon issue. With that this PR will be complete.

thalieht replied:
I have not been following the theming PRs so i don't know how they will play out. I think @Chocobo1 is right. Better keep changes in .cpp to a minimum in this PR (any renamed icons) and if i'm not mistaken this is the only thing remaining. Keep the new icons in the PR but undo any changes in this line, this has been a problem for a long time so i suppose it doesn't matter if it stays a little more.

Such GUI changes have to go through @sledgehammer999 but he is busy most of the time so i hope you won't be discouraged if it takes a while. 
Thanks for all this work.

now-im replied:
@thalieht I haven't changed anything in .cpp file regarding `error.svg`. I just shifted it from `skin` folder to `qbt-theme`. What should I do now?

thalieht replied:
Change this line back and put it back in `skin` because as you said, it breaks the icon in other places.

thalieht replied:
You should probably also revert the change a few lines below for the warning icon.

now-im replied:
@thalieht Reverted changes.

glassez commented:
Since #12861 is merged now you can provide your icons as "UI Theme" file (don't forget to update icons layout to follow the current one). Even if you want your icons to become built-in icon set it can help to review.

now-im commented:
@FranciscoPombal @glassez 
I am not a programmer. Can you give me some pointers? I will do the rest. Thanks.

FranciscoPombal commented:
I think this is what @glassez means: https://github.com/qbittorrent/qBittorrent/wiki/Create-custom-themes-for-qBittorrent

Though that page hasn't been updated to mention icons (at least Ctrl+F yields no results).
@jagannatharjun could you lend a hand here and also update the wiki page please?

now-im commented:
@glassez @FranciscoPombal 
Sorry Guys. I am not a programmer. I have never worked in UI or with any framework. I did this for fun. 
I had no idea you have combined the icons. As you can see I have messed the Pull as I had to pull from upstream and push my changes. I am not adept at this. After trying hours, I created a new pull with changes. 
Kindly merge my pull #12926, if possible. I will also make another with optimized icons. I just can't keep doing this. From next week lockdown of my country ends and I won't be able to spend any more time on this. It was a good learning experience for me. Thank you.
